src/cargo/util/toml.rs

```diff
@@ -561,6 +586,23 @@ impl TomlManifest {
     }
 }
 
+/// Will check a list of strings, and make sure they're unique within a vector. If not, the name of
+/// the offending binary is returned.
+fn are_unique_names(values: &Vec<String>) -> Result<(), String> {
+    use std::collections::HashSet;
+    let mut seen: HashSet<&String> = HashSet::new();
+    for v in values {
+        if seen.contains(&v) { return Err(v.to_string()) }
+        seen.insert(&v);
+    }
+    Ok(())
+}
+
+#[inline]
+fn collect_names(t: &Vec<TomlTarget>) -> Vec<String> {
```

Comment:
I think it'd be fine to remove this function and just have the function above take `Vec<TomlTarget>` instead